python/src/scipp/_shape.py

```diff
@@ -82,6 +82,7 @@ def concatenate(x: VariableLike, y: VariableLike, dim: str) -> VariableLike:
 
 
 def fold(x: VariableLike,
+         *,
```

Comment:
This means that the bound method must be called like
```.py
x.fold(dim='x', sizes=...)
```
meaning that all arguments are keyword only. I think this is consistent with what we discussed but it seems a bit odd especially since `dim` is always required.

Reply:
> I preferred the directional 'to'. We could alternatively use 'to' in both functions (in `fold` as a replacement for `sizes`).

So you would increase the asymmetry?
- `sc.fold(var, dim='x', to=['a', 'b'])
- `sc.flatten(var, dims=['a', 'b'], to='x', shape=[...])

Reply:
But `dim` is not the *only* arg that is always required. Also risks messing up with the `dims` argument, since order would be important. So I would actually require this here, yes.

Based on this, I just not that this works:
```python
import scipp as sc
sc.array(dims='ab', values=np.random.rand(2,2))
```
<img width="364" alt="image" src="https://user-images.githubusercontent.com/12912489/124934455-02b7f280-e005-11eb-9cd6-3e615ab48c3d.png">

Python is funny sometimes. :(

Reply:
Yep. Unfortunately, this is not even caught by type checkers, because `str` satisfies `Sequence[str]`...